# Lab | Intro to Probability

**Objective**

Welcome to this Intro to Probability lab, where we explore decision-making scenarios through the lens of probability and strategic analysis. In the business world, making informed decisions is crucial, especially when faced with uncertainties. This lab focuses on scenarios where probabilistic outcomes play a significant role in shaping strategies and outcomes. Students will engage in exercises that require assessing and choosing optimal paths based on data-driven insights. The goal is to enhance your skills by applying probability concepts to solve real-world problems.

**Challenge 1**

#### Ironhack Airlines 

Often Airlines sell more tickets than they have seats available, this is called overbooking. Consider the following:
- A plane has 450 seats. 
- Based on historical data we conclude that each individual passenger has a 3% chance of missing it's flight. 

If the Ironhack Airlines routinely sells 460 tickets, what is the chance that they have a seats for all passenger?

In [5]:
from scipy.stats import binom

n = 460
p = 0.97

binom_dist = binom (n, p)
prob_not_overbook = binom_dist.cdf(450)
round(prob_not_overbook, 3)

0.884

There is a 88.4% of chance that all the passengers will have a seat in the plane.

**Challenge 2**

#### Ironhack Call Center 

Suppose a customer service representative at a call center is handling customer complaints. Consider the following:
- The probability of successfully resolving a customer complaint on the first attempt is 0.3. 


What is the probability that the representative needs to make at least three attempts before successfully resolving a customer complaint?

In [6]:
from scipy.stats import geom

In [8]:
p = 0.30

geom_dist = geom(p)

prob_3_attemps = 1 - geom_dist.cdf(3)
round(prob_3_attemps, 3)

0.343

The probability is of 34.3%.

**Challenge 3**

#### Ironhack Website

Consider a scenario related to Ironhack website traffic. Where:
- our website takes on average 500 visits per hour.
- the website's server is designed to handle up to 550 vists per hour.


What is the probability of the website server being overwhelmed?

In [15]:
from scipy.stats import poisson

mu = 500

poisson_dist = poisson(mu)

prob_overwhelming = poisson_dist.pmf(550)
round(prob_overwhelming*100, 1)

0.2

There's a probability of 0.2%.

What is the probability of being overwhelmed at some point during a day? (consider 24hours)

In [17]:
prob_over_day = 1 - poisson_dist.cdf(550)
round(prob_over_day*100, 1)

1.3

There's a probability of 1.3%.

**Challenge 4**

#### Ironhack Helpdesk

Consider a scenario related to the time between arrivals of customers at a service desk.

On average, a customers arrives every 10minutes.

What is the probability that the next customer will arrive within the next 5 minutes?

In [19]:
from scipy.stats import expon

mu = 10

expon_dist = expon(scale = mu)

prob_next_5min = expon_dist.cdf(5)
round(prob_next_5min*100, 1)

39.3

There's a 39.3% chance of a customer arriving within the next 5 minutes.

-----------

If there is no customer for 15minutes, employees can that a 5minutes break.

What is the probability an employee taking a break?

In [21]:
prob_break = 1 - expon_dist.cdf(15)
round(prob_break*100, 1)

22.3

There's a 22.3% chance of an employee taking a break.

**Challenge 5**

The weights of a certain species of birds follow a normal distribution with a mean weight of 150 grams and a standard deviation of 10 grams. 

- If we randomly select a bird, what is the probability that its weight is between 140 and 160 grams?

In [22]:
from scipy.stats import norm

In [23]:
norm_dist = norm(loc = 150, scale = 10)

In [26]:
prob_weight = norm_dist.cdf(160) - norm_dist.cdf(140)
round(prob_weight*100, 1)

68.3

There's a 68.3% chance of the bird weighting between 140 and 160 g.

**Challenge 6**

If the lifetime (in hours) of a certain electronic component follows an exponential distribution with a mean lifetime of 50 hours, what is the probability that the component fails within the first 30 hours?

In [28]:
mu = 50

expon_dist = expon(scale = mu)

prob_failure = expon_dist.cdf(30)
round(prob_failure*100, 1)

45.1

There's a 45.1% chance that the component's failure will happen within the first 30 hours.